starlette/config.py

```diff
@@ -54,12 +54,18 @@ def __init__(
             self.file_values = self._read_file(env_file)
 
     def __call__(
-        self, key: str, cast: type = None, default: typing.Any = undefined
+        self,
+        key: str,
+        cast: typing.Union[type, typing.Callable] = None,
```

Comment:
I guess this could *just* be `cast: Callable` right? (Since types will be callable)

Reply:
@tomchristie Right you are. Updated the PR to reflect that fact.